common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -227,20 +227,22 @@ private boolean isBitrise() {
     }
 
     private void captureGitMetadata() {
-        buildScan.background(captureGitMetadataAction);
+        buildScan.background(new CaptureGitMetadataAction());
     }
 
-    private static final Action<BuildScanExtension> captureGitMetadataAction =
-        buildScan -> {
+    private static class CaptureGitMetadataAction implements Action<BuildScanExtension> {
```

Comment:
Can the class be final?

Reply:
I would pass Gradle and Providers to this class and store them as fields. This will avoid the null handling you had to introduce when querying env vars. I think...

Reply:
Passing Gradle would break configuration cache compatibility, but luckily we don't need it. And it appears that passing in ProvidersFactory does not break configuration-cache compatibility. So I made that change.

Reply:
Also marked the class as final.

Reply:
> And it appears that passing in ProvidersFactory does not break configuration-cache compatibility. 

Note that ProvidersFactory is meant as _the_ intended construct to maintain config cache compatibility. It's not a conincidence storing it as a field works with CC :-)